GVFS/GVFS.Platform.POSIX/POSIXPlatform.cs

```diff
@@ -76,9 +82,69 @@ public override void IsServiceInstalledAndRunning(string name, out bool installe
             throw new NotImplementedException();
         }
 
-        public override void StartBackgroundProcess(ITracer tracer, string programName, string[] args)
+        public override void StartBackgroundVFS4GProcess(ITracer tracer, string programName, string[] args)
         {
-            ProcessLauncher.StartBackgroundProcess(tracer, programName, args);
+            string programArguments = string.Empty;
+            try
+            {
+                programArguments = string.Join(" ", args.Select(arg => arg.Contains(' ') ? "\"" + arg + "\"" : arg));
+                ProcessStartInfo processInfo = new ProcessStartInfo(programName, programArguments);
+
+                // Redirecting stdin/out/err ensures that all standard input/output file descriptors are properly closed
+                // by dup2 before execve is called for the child process
+                // (see https://github.com/dotnet/corefx/blob/b10e8d67b260e26f2e47750cf96669e6f48e774d/src/Native/Unix/System.Native/pal_process.c#L381)
+                //
+                // However, this requires that the child process know that it needs to redirect its standard input/output to /dev/null and
+                // so this method can only be used with VFS4G processes that are aware they're being launched in the background
+                processInfo.RedirectStandardError = true;
+                processInfo.RedirectStandardInput = true;
+                processInfo.RedirectStandardOutput = true;
+
+                Process executingProcess = new Process();
+                executingProcess.StartInfo = processInfo;
+                executingProcess.Start();
+            }
+            catch (Exception ex)
+            {
+                EventMetadata metadata = new EventMetadata();
+                metadata.Add(nameof(programName), programName);
+                metadata.Add(nameof(programArguments), programArguments);
+                metadata.Add("Exception", ex.ToString());
+                tracer.RelatedError(metadata, "Failed to start background process.");
+                throw;
+            }
+        }
+
+        public override void PrepareProcessToRunInBackground()
+        {
+            int devNullIn = Open("/dev/null", (int)POSIXFileSystem.OpenFlags.O_RDONLY);
+            if (devNullIn == -1)
+            {
+                throw new Win32Exception(Marshal.GetLastWin32Error(), "Unable to open /dev/null for reading");
+            }
+
+            int devNullOut = Open("/dev/null", (int)POSIXFileSystem.OpenFlags.O_WRONLY);
```

Comment:
It's a shame we need this, but to double check remove the redirect to Null means we hang right?